sdks/python/container/boot.go

```diff
@@ -409,7 +409,7 @@ func installSetupPackages(ctx context.Context, logger *tools.Logger, files []str
 	if err := pipInstallPackage(ctx, logger, files, workDir, workflowFile, false, true, nil); err != nil {
```

Comment:
Do we want to do the same for installExtraPackages and pipInstallPackages?

Reply:
We can clean this up after the fact to a cleaner pattern, IIRC an issue similar to the one this fixes popped up when I was implementing the buffered logger originally which led to needing to handle the nil logger in the setup-only mode case.

Reply:
Thanks, @jrmccluskey ! Could you please  file an issue to follow up? thank you!